build.js

```diff
@@ -65,12 +79,28 @@ const injectAssets = (metafile) => {
   const scriptFile = Object.keys(outputs).find(file => file.endsWith('.js') && file.includes('ipfs-sw-index'))
   const cssFile = Object.keys(outputs).find(file => file.endsWith('.css') && file.includes('ipfs-sw-index'))
 
+  // Calculate hashes for script and CSS files
+  const scriptPath = scriptFile // The metafile outputs already contain absolute paths
+  const cssPath = cssFile // The metafile outputs already contain absolute paths
+  const scriptHash = calculateFileHash(scriptPath)
+  const cssHash = calculateFileHash(cssPath)
+
+  console.log(`Script hash (sha256): ${scriptHash}`)
+  console.log(`CSS hash (sha256): ${cssHash}`)
+
   const scriptTag = `<script type="module" src="${path.basename(scriptFile)}"></script>`
   const linkTag = `<link rel="stylesheet" href="${path.basename(cssFile)}">`
 
   // Read the index.html file
   let htmlContent = fs.readFileSync(htmlFilePath, 'utf8')
 
+  // Update the CSP header with the calculated hashes
+  const cspContent = `default-src 'none'; script-src 'self' 'sha256-${scriptHash}'; style-src 'self' 'sha256-${cssHash}'; img-src 'self'; font-src 'self'; manifest-src 'self'; connect-src 'self';`
+  htmlContent = htmlContent.replace(
+    /<meta http-equiv="Content-Security-Policy"[^>]*>/,
+    `<meta http-equiv="Content-Security-Policy" content="${cspContent}">`
```

Comment:
IF we set it here, and not on `ipfs.io` and `dweb.link` as well, we will be sabotaging all efforts to migrate people towards SW-based gateways, because "my website works on old gateway, and new one does not". 

We had long discussion in https://github.com/ipfs/in-web-browsers/issues/157 but we never pulled the trigger because PL prioritized `ipfs.io`   to "just work".


I'm happy to lock things down with CSP, it will fix phishing problem,  but we have to do it on all domains, not just `inbrowser.tld`.

Reply:
Doesn't the CSP configuration here only impact the SW gateway build itself? websites loaded with their own index.html should not get this treatment. Or am I missing something?

Reply:
If this is limited to initial HTML that bootstraps JS, what type of attack this protects us against?